static/js/components/email/hoc.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import R from 'ramda';
+
+import EmailCompositionDialog from './EmailCompositionDialog';
+import { getAllEmailProperties } from './lib';
+import { getDisplayName } from '../../util/util';
+
+export const withEmailDialog = R.curry((emailType, WrappedComponent) => {
+  class WithEmailDialog extends React.Component {
+    constructor(props) {
+      super(props);
+      this.renderCompositionDialog = this.renderCompositionDialog.bind(this);
+      this.allEmailProperties = getAllEmailProperties();
+      this.emailType = emailType;
+    }
+
+    renderCompositionDialog() {
+      let {
+        dispatch,
+        ui: { dialogVisibility },
+        email
+      } = this.props;
+      return <EmailCompositionDialog
+        dispatch={dispatch}
+        dialogVisibility={dialogVisibility}
+        email={email}
+        emailProperties={this.allEmailProperties}
+      />;
+    }
+
+    render() {
+      let emailDialog = this.props.emailDialog || this.renderCompositionDialog();
+
+      let emailOpenActions = this.props.emailOpenActions || {};
+      emailOpenActions[this.emailType] = R.partial(
+        this.allEmailProperties[this.emailType].emailOpenAction, [this.props.dispatch]
+      );
+
+      return <WrappedComponent
+        {...this.props}
+        emailDialog={emailDialog}
+        emailOpenActions={emailOpenActions}
+      />;
+    }
+  }
+
+  if (WithEmailDialog.displayName && !WithEmailDialog.displayName.startsWith('WithEmailDialogs')) {
+    WithEmailDialog.displayName = `WithEmailDialogs(${getDisplayName(WrappedComponent)})`;
+  }
+  return WithEmailDialog;
+});
```

Comment:
So to sum up I think this would be more straightforward if the API code (the methods on `EmailCompositionDialog` like `updateEmailFieldEdit` and `closeEmailComposerAndSend` were defined here. I think this would also simplify passing down the action dispatchers for opening the email and such things.